include/xsimd/types/xsimd_shim.hpp

```diff
@@ -0,0 +1,749 @@
+/***************************************************************************
+* Copyright (c) 2016, Wolf Vollprecht, Johan Mabille and Sylvain Corlay    *
+*                                                                          *
+* Distributed under the terms of the BSD 3-Clause License.                 *
+*                                                                          *
+* The full license is in the file LICENSE, distributed with this software. *
+****************************************************************************/
+
+#ifndef XSIMD_SHIM_HPP
+#define XSIMD_SHIM_HPP
+
+#include <cmath>
+
+#include "xsimd_base.hpp"
+
+namespace xsimd
+{
+
+    namespace detail 
+    {
+        template <class T>
+        struct get_int_type
+        {
+            using signed_type = typename std::make_signed<T>::type;
+            using unsigned_type = typename std::make_unsigned<T>::type;
+        };
+
+        template <>
+        struct get_int_type<float>
+        {
+            using signed_type = int32_t;
+            using unsigned_type = uint32_t;
+        };
+
+        template <>
+        struct get_int_type<double>
+        {
+            using signed_type = int64_t;
+            using unsigned_type = uint64_t;
+        };
+
+        template <class T>
+        struct get_float_type;
+
+        template <>
+        struct get_float_type<int32_t>
+        {
+            using type = float;
+        };
+
+        template <>
+        struct get_float_type<int64_t>
+        {
+            using type = double;
+        };
+    }
+
+    /*************************
+     * batch_bool<T, N> *
+     *************************/
+
+    template <class T, std::size_t N>
+    struct simd_batch_traits<batch_bool<T, N>>
+    {
+        using value_type = typename detail::get_int_type<T>::unsigned_type;
+        static constexpr std::size_t size = N;
+        using batch_bool_type = batch_bool<T, N>;
+    };
+
+    template <class T, std::size_t N>
+    class batch_bool : public simd_batch_bool<batch_bool<T, N>>
```

Comment:
This should be a specialization of `batch_bool` for `N=1`, we should not provide generic implementaion of `batch_bool` for arbitrary `N`.

Reply:
I was thinking the following: first, we could easily test int64_t on NEON 32bit devices, where double is NOT available. 

Also, we can easily prevent some implementations for e.g. batch<uint8_t, 16> and hope (or actually measure!) that compilers are able to autovectorize the code (especially if we provide some hints, such as `__assume_aligned` etc. -- and then later we can swap out/specialize the implementations with the real stuff.